tbot_contrib/gpio.py

```diff
@@ -0,0 +1,56 @@
+import tbot
+from tbot.machine import linux
+
+
+class Gpio:
+    def __init__(self, host: linux.LinuxShell, gpio_number: int):
+
+        self.host = host
+        self._gpio_number = str(gpio_number)
+        self._gpio_sysclass_path = self.host.fsroot / "sys/class/gpio"
+        self._gpio_path = self._gpio_sysclass_path / f"gpio{self._gpio_number}"
+        self._export()
+
+    def _export(self) -> None:
+        if not self._gpio_path.is_dir():
+            self.host.exec0(
+                "echo",
+                str(self._gpio_number),
+                linux.RedirStdout(self._gpio_sysclass_path / "export"),
+            )
+
+    def _is_in_direction(self) -> bool:
+        direction = self.host.exec0("cat", self._gpio_path / "direction")
+        return direction.strip().lower() == "in"
+
+    def set_in_direction(self) -> None:
+        if self._is_in_direction():
+            pass
+        else:
+            self.host.exec0(
+                "echo", "in", linux.RedirStdout(self._gpio_path / "direction")
+            )
+
+    def set_out_direction(self) -> None:
+        if not self._is_in_direction():
+            pass
+        else:
+            self.host.exec0(
+                "echo", "out", linux.RedirStdout(self._gpio_path / "direction")
+            )
+
+    def get_direction(self) -> str:
+        return self.host.exec0("cat", self._gpio_path / "direction").strip()
```

Comment:
Please use [`Path.read_text()`](https://tbot.tools/modules/machine_linux.html#tbot.machine.linux.Path.read_text) and [`Path.write_text()`](https://tbot.tools/modules/machine_linux.html#tbot.machine.linux.Path.write_text) where possible.